### PR TITLE
ID-55 select2 required

### PR DESCRIPTION
**Pull request by johnhuangclever**
Probably helpful to review commit-by-commit

**Overview:**
Support a "required" and error state. Summary of changes:
* Move the `TextInput2Requirement` enum to more generalized form utils
* add requirement option and requirement label elements to Select2
* Support error states for required prop (error and initialError)
* open menu when clear button is clicked

**Screenshots/GIFs:**

https://user-images.githubusercontent.com/13126257/109743707-ff5bb880-7b85-11eb-98e0-5a4d6d0b8c44.mov

**Testing:**

- [ ] Unit tests
- Manual tests:
  - [x] Chrome
  - [x] Safari
  - [ ] IE11

**Roll Out:**

- Before merging:
  - [x] Updated docs
  - [ ] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add an entry in `ComponentsView.componentsToDisplay` using this template:
      ```
      {
        componentLink: "<COMPONENT LINK>",
        componentImg: "<COMPONENT LINK>.png",
        componentName: "<COMPONENT NAME>",
        componentImgAlt: "A <COMPONENT NAME> component",
      },
      ```
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/Select2/Select2.less [hunk 1]
```diff
@@ -94,15 +115,23 @@
...
 
+.Select2--trailingElement {
+  .margin--left--xs();
+  min-height: @size_2xl;
+}
+
+.Select2--errorIcon {
+  color: @alert_red;
+  display: flex;
+  align-items: center;
```
Comment (prime-time) [inline src/Select2/Select2.less:126]: I notice that the flex mixins are used in a rule at the top of the file. Is there a particular reason for using native css flex here? I'm not suggesting one over the other, rather just wondering if it was intentional.

File: src/Select2/Select2.less [hunk 2]
```diff
@@ -48,10 +62,22 @@
...
+  }
+
   /* stylelint-disable-next-line no-descending-specificity */
   &:hover {
     .border--m(@primary_blue);
   }
+
+  &--error:hover {
+    .border--m(@alert_red);
+  }
```
Comment (prime-time) [inline src/Select2/Select2.less:76]: I'm not 100% sure, but I think you could combine this with the rule a few lines above like this.
```suggestion
  &--error, &--error:hover {
    .border--m(@alert_red);
  }
```

File: src/Select2/Select2.tsx [hunk 3]
```diff
@@ -111,20 +129,31 @@ const Select2: React.FC<Props> = ({
...
-      </label>
+      <div className={classNames(cssClass.INFO_ROW, hideLabel && cssClass.INFO_ROW_LABEL_HIDDEN)}>
+        <label
+          className={classNames(cssClass.LABEL, hideLabel && cssClass.LABEL_HIDDEN)}
+          {...getLabelProps()}
+        >
+          {label}
+        </label>
+        {requirement && (
+          <label className={cssClass.INFO_REQUIREMENT} aria-live="polite" {...getLabelProps()}>
```
Comment (prime-time) [inline src/Select2/Select2.tsx:147]: Woa interesting, I didn't know `aria-live="polite" ` was a thing.

Comment (johnhuangclever) [reply src/Select2/Select2.tsx:147]: Copied this from the `TextInput` component. Honestly I don't know how it fulfills a11y requirements https://github.com/Clever/components/blob/b54fd68cafa664519ca199853beb8554dc812966/src/TextInput/TextInput.tsx#L234 🤷‍♀️

File: src/Select2/Select2.less (hunk 1 above)
Comment (johnhuangclever) [reply src/Select2/Select2.less:126]: whoops missed it, yeah. Thanks.

We need the flex mixins for cross-browser support. I think fampo set it up so we could use the native attributes and to autogenerate the platform-specific attributes

(we should use that at some point)

File: src/Select2/Select2.less (hunk 2 above)
Comment (johnhuangclever) [reply src/Select2/Select2.less:76]: I tried using commas in alot of places and they didn't seem to work. Not sure if it's my css problem or a less quirk or something, but I couldn't get it to work

File: src/Select2/Select2.less (hunk 1 above)
Comment (prime-time) [reply src/Select2/Select2.less:126]: Ah that's right, I forgot what the mixins did besides save us from typing a few characters 😄

File: src/Select2/Select2.less (hunk 2 above)
Comment (prime-time) [reply src/Select2/Select2.less:76]: Sounds like I need to review selectors 🦧 . Carry on.

**Review by prime-time — APPROVED** [review]
Niiice, looks good to me. 

You probably also want to bump the version.

**Pull request merged**